From 86b6b0651309e9a8cb0ed56c6f76ed2a206f0cc3 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Mon, 29 Jan 2024 00:43:09 -0500
Subject: [PATCH] 2

---
 net/batman-adv/netlink.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/batman-adv/netlink.c b/net/batman-adv/netlink.c
index 238825d..5bbd246 100644
--- a/net/batman-adv/netlink.c
+++ b/net/batman-adv/netlink.c
@@ -1266,7 +1266,11 @@ batadv_get_vlan_from_info(struct batadv_priv *bat_priv, struct net *net,
  *
  * Return: 0 on success or negative error number in case of failure
  */
+#if LINUX_VERSION_IS_GEQ(6, 6, 0)
+static int batadv_pre_doit(const struct genl_split_ops *ops, struct sk_buff *skb,
+#else
 static int batadv_pre_doit(const struct genl_ops *ops, struct sk_buff *skb,
+#endif
 			   struct genl_info *info)
 {
 	struct net *net = genl_info_net(info);
@@ -1335,7 +1335,11 @@ err_put_softif:
  * @skb: Netlink message with request data
  * @info: receiver information
  */
+#if LINUX_VERSION_IS_GEQ(6, 6, 0)
+static void batadv_post_doit(const struct genl_split_ops *ops, struct sk_buff *skb,
+#else
 static void batadv_post_doit(const struct genl_ops *ops, struct sk_buff *skb,
+#endif
 			     struct genl_info *info)
 {
 	struct batadv_hard_iface *hard_iface;
-- 
2.17.1

